src/execute/index.js

```diff
@@ -25,6 +25,52 @@ import { isOpenAPI3 } from '../helpers/openapi-predicates.js';
 
 const arrayOrEmpty = (ar) => (Array.isArray(ar) ? ar : []);
 
+const findObjectSchema = (schema, { recurse = true, depth = 1 } = {}) => {
+  if (!isPlainObject(schema)) return undefined;
+
+  // check if the schema is an object type
+  if (schema.type === 'object' || (Array.isArray(schema.type) && schema.type.includes('object'))) {
+    return schema;
+  }
+
+  if (depth > TRAVERSE_LIMIT) return undefined;
+
+  if (recurse) {
+    // traverse oneOf keyword first
+    const oneOfResult = Array.isArray(schema.oneOf)
+      ? schema.oneOf.find((subschema) => findObjectSchema(subschema, { recurse, depth: depth + 1 }))
+      : undefined;
+
+    if (oneOfResult) return oneOfResult;
+
+    // traverse anyOf keyword second
+    const anyOfResult = Array.isArray(schema.anyOf)
+      ? schema.anyOf.find((subschema) => findObjectSchema(subschema, { recurse, depth: depth + 1 }))
+      : undefined;
+
+    if (anyOfResult) return anyOfResult;
+  }
+
+  return undefined;
+};
+
+const parseJsonValue = ({ value, shouldFailSafe = false }) => {
```

Comment:
NITPICK: Can I suggest naming this `parseJsonObject` ? That's what this function basically does. And maybe consider naming `shouldFailSafe` as `shouldFailSilently` or just `silentFail` ?

```suggestion
const parseJsonObject = ({ value, silentFail = false }) => {
```